### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by JayakumarPawan**
Romeo Juliet Ocean Puzzle

Added a copy of the ocean scene that implements the r+j ocean puzzle where one has to deliver a bottle from top left to bot right tile in 2 moves using the anchor. 
## Description

updated the msg in bottle prefab to have a better animator, hitbox, and a script that lets it sink then destroy itself.
configured romeos dialogue to support this puzzle
created a bottle manager which conditionally ccreates bottles and evaluatates whether the puzzle has been solved
added a rose object and configured juliets dialogue 
configured the shop tavern scripts to handle a new thing to turn in (the rose)
## Related Task

r+j ocean pzl
## How Has This Been Tested?
checked if player can run over bottle and if splash particles are on diff layer
checked if bottle sinks after 3 moves and is deleted properly
checked if puzzle can be retried
checked if puzzle only lets u use the 3 correct carrier tiles
checked if bottle moves and animates properly
checked if juliet gives rose properly
checked if tavern keep accepts rose

### CODE REVIEW

File: Slider/Assets/Scripts/Map/Ocean/BottleManager.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                Debug.Log("invalid tile");
+                DestroyBottle();
+            }
+
+
+            
+        }
+    }
+
+    public void InvalidTile(Condition c)
```
Comment (randomerz) [inline Slider/Assets/Scripts/Map/Ocean/BottleManager.cs:98]: Is this needed/used? NPCs have a dcond that specifically checks grid regex alr

File: Slider/Assets/Scripts/Map/Ocean/BottleManager.cs [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+public class BottleManager : MonoBehaviour
+{
+    // set in the editor
+    public GameObject bottle_prefab; 
+
+    private int turncounter = 0;
+    private GameObject bottle = null;
+    private bool puzzleActive = false;
+    private bool puzzleSolved = false;
```
Comment (randomerz) [inline Slider/Assets/Scripts/Map/Ocean/BottleManager.cs:13]: `puzzleSolved` probably needs to be serialized so it's saved between playthroughs, see `SaveSystem.current.SetBool(string, bool)` (and do a `GetBool(str, bool)` in awake)

File: Slider/Assets/Scripts/Sliders/Grid/SGridAnimator.cs [hunk 3]
```diff
@@ -118,46 +120,23 @@ protected IEnumerator StartMovingAnimation(STile stile, Movement moveCoords, SMo
...
-    // DC: this is a lot of parameters :)
-    protected IEnumerator DisableBordersAndColliders(
-        STile[,] grid, 
-        SGridBackground[,] bgGrid, 
-        HashSet<Vector2Int> positions, 
-        Dictionary<Vector2Int, List<int>> borders,
-        List<Coroutine> moveCoroutines
-        )
+
+    protected IEnumerator DisableBordersAndColliders(STile[,] grid, SGridBackground[,] bgGrid, HashSet<Vector2Int> positions, Dictionary<Vector2Int, List<int>> borders, List<Coroutine> moveCoroutines)
```
Comment (randomerz) [inline Slider/Assets/Scripts/Sliders/Grid/SGridAnimator.cs:139]: i put the params line by line cause i think it's easier to read that way 💀

File: Slider/Assets/Scripts/Map/Ocean/MessageInBottleController.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+using UnityEngine;
+
+public class MessageInBottleController : MonoBehaviour
+{
+
+    public void DestroyBottle()
```
Comment (randomerz) [inline Slider/Assets/Scripts/Map/Ocean/MessageInBottleController.cs:6]: make sure to remove its uitracker too

**Review by randomerz — COMMENTED** [review]
Overall changes look good! Nothing major to report, just some nitpicks about how stuff is handled. Playing through it in game feels good too, with the movement + ui showing up

Other things to change
- maybe make the bottle sink happen as the move from 2->3 is going on, rather than once move 2->3 is completed
- the bottle incoming animation has a jitter in one of the frames
- make sure bottle is destroyed/poofed when juliet reads it
- when juliet gives the rose, don't do rose.setActive in the NPC conds (`oceanGrid.ActivateCollectible` should handle it)
- Fix her juliet 3rd line of dialogue
- make sure to revert changes to `Items.png` since you added the new `rose.png` sprite
- make sure all changes get pushed to ocean scene

Things for the future to consider,
- maybe the entrance animation should be a bit more dramatic so that they notice the bottle coming in

File: Slider/Assets/Scripts/Map/Ocean/BottleManager.cs (hunk 1 above)
Comment (JayakumarPawan) [reply Slider/Assets/Scripts/Map/Ocean/BottleManager.cs:98]: That's true but its convenient if we want to change the set of valid tiles later (now that i made a string with all the valid tiles)

**Pull request closed without merging**